R/ncdf.R

```diff
@@ -377,8 +385,10 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
     XY_curvi = unlist(cvar[c("X", "Y")])
     
     if (all(!is.na(XY_curvi)) && all(variables$ndims[match(XY_curvi, variables$name)] == 2)) {
-      curvilinear = XY_curvi
+      return(XY_curvi)
     }
+  } else {
+    return(character(0))
   }
```

Comment:
should the top level else condition give an error? Otherwise, NULL is returned.

Reply:
Added a return `character(0)` and cleaned up the function a bit more.